compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -1907,43 +1907,58 @@ setTransposeLikeOpRootConfig(mlir::FunctionOpInterface entryPointFn,
-  if (!hasAVX2Feature(targetAttr) ||
-      !x86TransposeLoweringPrecondition(genericOp)) {
+  bool targetIsAArch64SME = isAArch64(targetAttr) &&
+                            clEnableScalableVectorization &&
+                            hasSMEFeature(targetAttr);
+  if ((!hasAVX2Feature(targetAttr) && !targetIsAArch64SME) ||
+      !transposeLoweringPrecondition(genericOp))
     return failure();
-  }
 
   DistributionHeuristicConfig distConfig;
-  distConfig.minTileSizes = getMinTilingSizesForEachDim(
-      entryPointFn, genericOp, linalgOpInfo, targetMLTransInfo);
-  if (llvm::count_if(distConfig.minTileSizes,
-                     [](int64_t tileSize) { return tileSize > 1; }) != 2) {
-    // Transpose patterns are not applicable if vectorizing more or less than
-    // two dims.
-    return failure();
-  }
+  if (hasAVX2Feature(targetAttr)) {
+    distConfig.minTileSizes = getMinTilingSizesForEachDim(
+        entryPointFn, genericOp, linalgOpInfo, targetMLTransInfo);
+    if (llvm::count_if(distConfig.minTileSizes,
+                       [](int64_t tileSize) { return tileSize > 1; }) != 2) {
+      // Transpose patterns are not applicable if vectorizing more or less than
+      // two dims.
+      return failure();
+    }
 
-  // Make sure that the original tile sizes are greater than or equal to the
-  // tile sizes to be used for the transpose op (e.g., 8x8, 16x16, etc).
-  int64_t targetVectorSize = 8;
-  if (llvm::any_of(distConfig.minTileSizes, [&](int64_t tileSize) {
-        return tileSize > 1 && tileSize < targetVectorSize;
-      })) {
-    return failure();
-  }
+    // Make sure that the original tile sizes are greater than or equal to the
+    // tile sizes to be used for the transpose op (e.g., 8x8, 16x16, etc).
+    int64_t targetVectorSize = 8;
+    if (llvm::any_of(distConfig.minTileSizes, [&](int64_t tileSize) {
+          return tileSize > 1 && tileSize < targetVectorSize;
+        })) {
+      return failure();
+    }
 
-  // Target 16x16 tile sizes if there are AVX512 features and all the tile sizes
-  // are greater than or equal to 16.
-  if (hasAVX512fFeature(targetAttr) &&
-      llvm::all_of(distConfig.minTileSizes, [](int64_t tileSize) {
-        return tileSize == 1 || tileSize >= 16;
-      })) {
-    targetVectorSize = 16;
-  }
+    // Target 16x16 tile sizes if there are AVX512 features and all the tile
+    // sizes are greater than or equal to 16.
+    if (hasAVX512fFeature(targetAttr) &&
+        llvm::all_of(distConfig.minTileSizes, [](int64_t tileSize) {
+          return tileSize == 1 || tileSize >= 16;
+        })) {
+      targetVectorSize = 16;
+    }
+
+    // Replace dims to be vectorized with the new tile sizes.
+    std::replace_if(
+        distConfig.minTileSizes.begin(), distConfig.minTileSizes.end(),
+        [](int64_t tileSize) { return tileSize > 1; }, targetVectorSize);
+  } else if (targetIsAArch64SME) {
+    auto elementType = nonWideningLinalgElementType(genericOp);
```

Comment:
I wonder if this should be more restrictive? This seems to set 2-D scalable tile sizes for any transpose-like operation, but not all of them can/do lower to ArmSME, for example, I tired out:
```
  %empty = tensor.empty(): tensor<2x16x16xf32>
  %rhs = linalg.transpose ins(%transposed_b : tensor<2x16x16xf32>)
      outs(%empty : tensor<2x16x16xf32>) permutation = [0, 2, 1]
```
...it set 2-D scalable flags but ended up unvectorized (and used linalg-to-loops in the end).

Reply:
absolutely, I misinterpreted `x86TransposeLoweringPrecondition` to be restricted to 2 dims. I've kept that as is and created a separate one for SME to be more restrictive, test also added.